packages/jira-adapter/src/change_validators/workflowsV2/status_mappings.ts

```
@@ -0,0 +1,240 @@
+  const issueTypeReferences = await getWorkflowSchemeMigrationIssueType({
+    workflowSchemeInstance,
+    workflowName,
+    elementsSource,
+  })
+  if (_.isEmpty(issueTypeReferences)) {
+    return []
+  }
+  const statusMappings = projectReferences.flatMap(project =>
+    issueTypeReferences.map(issueType =>
+      createStatusMapping({
+        project,
+        issueType,
+        statusesToMigrate: removedStatuses,
+      })))
+  const missingStatusMappings = _.differenceWith(statusMappings, existingStatusMappings, isSameStatusMappings)
+  if (_.isEmpty(missingStatusMappings)) {
+    return []
+  }
+  return statusMappings
+}
+
+const getStatusMappingsFormat = (statusMappings: StatusMapping[]): string =>
+  `statusMappings = [\n    ${statusMappings.map(statusMapping => createStatusMappingStructure(statusMapping)).join('\n    ')}\n]`
+
+export const workflowStatusMappingsValidator: ChangeValidator = async (changes, elementsSource) => {
+  const relevantChanges = getRelevantChanges(changes)
+  if (elementsSource === undefined || _.isEmpty(relevantChanges)) {
+    return []
+  }
+  const projects = await getInstancesFromElementSource(elementsSource, [PROJECT_TYPE])
+  // all the active workflowSchemes
+  const workflowSchemeNameToInstance = await awu(projects)
+    .filter(projectHasWorkflowSchemeReference)
+    .map(project => project.value.workflowScheme.getResolvedValue(elementsSource))
+    .filter(isInstanceElement)
+    .keyBy(instance => instance.elemID.getFullName())
+
+  const workflowSchemeNameToProjectReferences = _.groupBy(
+    projects.filter(projectHasWorkflowSchemeReference)
+      .map(project => new ReferenceExpression(project.elemID, project)),
+    project => project.value.value.workflowScheme.elemID.getFullName(),
+  )
+  return awu(relevantChanges).flatMap(async change => {
+    const workflowInstance = getChangeData(change)
+    const existingStatusMappings = isStatusMappings(workflowInstance.value.statusMappings)
```

maybe if `statusMappings` exists but it's not `isStatusMappings`, we should return a different error, saying that the syntax is wrong (maybe using the scheme guard result?), otherwise I'm not sure that users will understand why it's not working for them.